### DISCUSSION TITLE
Idea to skip an update if no code changes are part of the update

### DISCUSSION THREAD

**Discussion by HeavyWombat** [renovatebot/renovate#29395] [Suggest an Idea]
### Tell us more.

In the context of Renovate performing Go module updates to create a PR, there are libraries that actually do releases where there might be no change in the actual Go source code. The PR from Renovate then only contains an update in `go.mod` with the new version, an update in `go.sum`, and optionally an entry in the `vendor/modules.txt`.

My question would be: Could there be some kind of enhancement to detect updates that do not include source code updates and therefore would be skipped to reduce noise and unnecessary commits?

Thanks in advance for your feedback.

**Comment by rarkins** [comment]
How/where would Renovate determine that no source code was changed? e.g. based on vendor/ content?

**Reply by HeavyWombat** [reply]
Now that I think about it, yes, I only focused on having a `vendor` directory and with that it is relatively easy to conclude that there is no Go source code update.

**Reply by rarkins** [reply]
How common is it that a Go library publishes a new version without any code changes?

**Reply by HeavyWombat** [reply]
Not very common, it is an edge case I think. In my case, it is probably due to some automation on the library side that produce releases with any change, where the change is not necessarily in the code, but in deployment configuration or similar.

**Reply by rarkins** [reply]
It's not an _easy_ change for Renovate, because what it means in practice is to gracefully abort an update while still not confusing users. e.g. the dashboard and logs will show that an update is needed, but the PR won't appear, and probably the users would need to look at logs to understand why.

So for now I'll leave this as an Idea discussion as I'm not sure how to make this work well, or if it's highly demanded.

**Reply by HeavyWombat** [reply]
Thanks for your feedback. I can totally understand the rational, it was just a passing thought when I saw such a no-up update. On that topic, is there any way to have a hook to run a snippet of code as part of the update to tell it to gracefully abort/skip what it is currently doing?

**Reply by rarkins** [reply]
No we don't currently have such a capability
